### ISSUE TITLE
Unable to Start 4.0.1 wallet

### ISSUE THREAD

**Issue by knight9175** [vergecurrency/VERGE#563]
# Issue Template

## Context

I am trying to start the 4.0.1 wallet but am unable to.  when I run the verge-qt.exe file I get a verge splash screen then a error window then the wallet exits. 
I've run  netstat -an | find "9089" but it does not find anything on that port. 
I have run normally as well as Run as Admin with the same results. 

This computer had an old install of Verge 3.0 QT Wallet. 

## Prerequisites

* [y ] Are you running the latest version?

## User Story:

* "I'm submitting a ... "
  - [*] bug report
  - [ ] feature request
  - [ ] support request 
  - [ ] translation request 

Os : Windows 10

## Expected Behavior
I expect the wallet to start when running the Verge-Qt.exe file. 

## Current Behavior

I am trying to run the 4.0.1 wallet - latest from release repository and it errors out with a message : Timed out waiting for onion hostname
Wallet Exits

## Screenshots

![vergetimeout](https://user-images.githubusercontent.com/35758806/35427809-7603a7d2-0220-11e8-8dc2-3ba5446e3313.PNG)

## Possible Solution

please make the install process a litle more like 3.0 release where it just worked. 

## Steps to Reproduce (for bugs)

I don't know as not everyone is experiencing this. but these are my steps

* Steps :
1. download 4.0.1 fork for stealth
2. Unzip file into a folder
3.Run Verge-qt.exe
4. Get Error

## Your Environment

* Verge Release: 4.0.1
* CPU: i7
* GPU: 1080Ti x2
* RAM: 32BG 
* OS: Windows 10 Pro
    - [ ] I'm using Mac
    - [ ] I'm using Linux
    - [ ] I'm using Mac

## Any extra information that might be useful in the debugging process.
Db.log file was empty - 
this is the end of the debug.log file. 

VERGE version v4.0.1.0-794660a-funky-beta (2018-01-19 03:31:22 -0500)
Using OpenSSL version OpenSSL 1.0.2k  26 Jan 2017
Startup time: 01/26/18 06:34:11
Default data directory C:\Users\Alireza\AppData\Roaming\VERGE
Default mining hash algorithm: scrypt
Used data directory C:\Users\Alireza\AppData\Roaming\VERGE
scrypt: using scrypt-sse2 as built.
dbenv.open LogDir=C:\Users\Alireza\AppData\Roaming\VERGE\database ErrorFile=C:\Users\Alireza\AppData\Roaming\VERGE\db.log
Bound to 127.0.0.1:21102
TOR thread started.
Using OBFS4.
No onion hostname yet, will retry in 2 seconds... (1/8)
No onion hostname yet, will retry in 2 seconds... (2/8)
No onion hostname yet, will retry in 2 seconds... (3/8)
No onion hostname yet, will retry in 2 seconds... (4/8)
Onion thread exited.No onion hostname yet, will retry in 2 seconds... (5/8)
No onion hostname yet, will retry in 2 seconds... (6/8)
No onion hostname yet, will retry in 2 seconds... (7/8)
No onion hostname yet, will retry in 2 seconds... (8/8)

**Comment by knight9175** [comment]
hope this better.

**Comment by Max-Meldrum** [comment]
Is the Windows firewall or something similar activated?

**Comment by justinvforvendetta** [comment]
do you have tor already running somewhere else? also are you in china?

**Comment by jhmblvd** [comment]
I have the exact same problem. I cannot access the wallet bc this dialog box opens with Timed Out waiting for Tor hostname. Checked my Firewall Vergeqt is allowed , Tor is allowed, See nothing that would cause this. I am on Windows 10 as well and cannot get this working. Help or a fix would be appreciated.

**Comment by Max-Meldrum** [comment]
@jhmblvd Another user had another crypto wallet that was also using port 9089, do u have another wallet running? Or do you pehaps have vmware installed on computer?

**Comment by knight9175** [comment]
I am in Seattle, was. Firewall is disabled. I had this problem on a Linksys
router and now I am on new metering equipment and still have this problem.

**Comment by michpiatek** [comment]
Do you have tor/onion directory structure in your wallet directory? It should be created at the first run and contain hostname and private_key files. If they are not present that might be because of insufficient permissions to create them when running as normal user. Try as administrator or modify permissions. Not a windows guy though, just guessing.

**Comment by Max-Meldrum** [comment]
Might be a long shot and might not be related to this specific issue but perhaps worth to look at:
"Check your system clock. If it's more than a few hours off, Tor will refuse to build circuits. For Microsoft Windows users, synchronize your clock under the clock -> Internet time tab. In addition, correct the day and date under the 'Date & Time' Tab. Also make sure your time zone is correct"

**Comment by knight9175** [comment]
These are all fine on my system. Maybe better logging or a verbose option
could help. Can I run the verge-qt in a more verbose mode ?

**Comment by Max-Meldrum** [comment]
@jhmblvd  @knight9175 The error popup occurs as the code cannot find the file C:\Users\Username\AppData\Roaming\VERGE/tor/onion/hostname , but can you check if you have a tor.log file here C:\Users\Username\AppData\Roaming\VERGE/tor/tor.log ?

Some clues could be in there.

**Comment by michpiatek** [comment]
@knight9175 did you check the folders I wrote about? That's a first step, if you don't have them then the whole conversation is pointless. If you have them but hostname file isn't there, my previous message also applies. So check that first please.

**Comment by knight9175** [comment]
Is hostname a directory or a file

**Comment by knight9175** [comment]
Do I need to create that directory or Phil that file with something. Is
this a party of install directions in missed ?

**Comment by Max-Meldrum** [comment]
hostname is a file. the tor directy should be created when executing the wallet for first time etc...

**Comment by knight9175** [comment]
I know I made that directory up to onion. No files are in there. I will
double check tonight when I go home.

On Jan 26, 2018 11:52 AM, "Alireza Razmpoosh" <kni8.2k@gmail.com> wrote:

Do I need to create that directory or Phil that file with something. Is
this a party of install directions in missed ?

**Comment by Max-Meldrum** [comment]
This is how the tor.log looks like after an successful start.
![torlog](https://user-images.githubusercontent.com/11488530/35458229-d48d4842-02db-11e8-8396-634677262a47.png)

**Comment by knight9175** [comment]
Running wallet for first time did not create directories. Not did it ask
for permission to create. Even when started as admin. I created directories
manually.

**Comment by michpiatek** [comment]
@knight9175 

Right click on C:\Users\Username\AppData\Roaming\ folder, properties, security and make sure that your user has full control.

**Comment by jhmblvd** [comment]
I do have vTorrent installed. I also have VMWARE Desktop Pro installed but not running.

**Comment by michpiatek** [comment]
@jhmblvd Please read my last message here.

**Comment by knight9175** [comment]
I have said it before that I do not have those files in that directory. I
will check permissions tonight. But those files were never made for me.

**Comment by jhmblvd** [comment]
This is what I get when i try this:

Jan 26 15:56:05.761 [notice] Tor 0.3.2.8-rc-dev running on Very recent
version of Windows [major=10,minor=0] with Libevent 2.0.22-stable, OpenSSL
1.0.2k, Zlib 1.2.11, Liblzma N/A, and Libzstd N/A.
Jan 26 15:56:05.762 [notice] Tor can't help you if you use it wrong! Learn
how to be safe at https://www.torproject.org/download/download#warning
Jan 26 15:56:05.763 [notice] Configuration file "C:\Users\Valued
Customer\AppData\Roaming\VERGE\tor\torrc" not present, using reasonable
defaults.
Jan 26 15:56:05.978 [warn] Path for GeoIPFile (<default>) is relative and
will resolve to D:\John\Documents\CryptoWallets\VERGE-windows\<default>. Is
this what you wanted?
Jan 26 15:56:05.978 [warn] Path for GeoIPv6File (<default>) is relative and
will resolve to D:\John\Documents\CryptoWallets\VERGE-windows\<default>. Is
this what you wanted?
Jan 26 15:56:05.986 [notice] Scheduler type KISTLite has been enabled.
Jan 26 15:56:05.986 [notice] Opening Socks listener on 127.0.0.1:9089
Jan 26 15:56:05.986 [warn] Could not bind to 127.0.0.1:9089: Address
already in use [WSAEADDRINUSE ]. Is Tor already running?
Jan 26 15:56:05.987 [warn] Failed to parse/validate config: Failed to bind
one of the listener ports.
Jan 26 15:56:05.987 [err] Reading config failed--see warnings above.
VERGE: Timed out waiting for onion hostname.


John Mahoney

**Comment by Max-Meldrum** [comment]
@jhmblvd 
Either some other service is using port 9089 or it is just "stuck".  Try to follow the accepted answer here with port 9089.:  https://stackoverflow.com/questions/39632667/how-to-kill-a-currently-using-port-on-localhost-in-windows 

And then try again. ps: I don't know if you are running some kind of important application that is using port 9089, so I'm not responsible if you disrupt something in another program etc..

**Comment by jhmblvd** [comment]
Nope, doesn't work. Nothing running on that port, stopped it anyway, still
get the same error. Still not seeing Tor folder logs being created.

John Mahoney

**Comment by jrsikors** [comment]
Just for grins -- try running it from a command prompt window or bat file:
command prompt:  verge-qt.exe -datadir=c:\vergetest\data\
bat file:  "verge-qt.exe" -datadir=c:\vergetest\data\

I just tested with 4.0.1 fork for stealth and it created a new \tor\onion folder , with a hostname and private_key file

**Comment by jrsikors** [comment]
that was for @knight9175  , but might not hurt for @jhmblvd  to try too.

@jhmblvd  I'd also try temporarily disabling anti virus...  I don't know you meant by "nothing running on that port, stopped it anyway".  What did you stop if you didn't find anything running?

**Comment by jhmblvd** [comment]
Not working for me on Windows 10 machine. But  I was able to install on my
Windows 7 box so I'll use that for now. I have it up and running on the W7
machine and it is syncing the blockchain. How do I restore the wallet from
the backup .dat file from the Windows 10 box  at this point? It is still
syncing, when it's finished syncing what do I need to do in order to import
the backup of the wallet?
Thank you.
John .

John Mahoney

**Comment by justinvforvendetta** [comment]
im running it on my windows 10 box. question, @jhmblvd is your windows 10 a 32 or 64bit?

to restore the wallet.dat, just go into the %appdata% folder, find verge, and copy the wallet.dat to the same location on the other computer (while the wallet is NOT running)

**Comment by jhmblvd** [comment]
I'm on a 64  Bit version of Windows 10. I fired it up using my W7 machine
and had no issues. So, something on my W10 system was blocking the TOR
browser.
J

John Mahoney

**Comment by justinvforvendetta** [comment]
yeah thats what it sounds like. and you arent running a vm, or another instance of tor on the windows 10 machine?

**Comment by Technophant** [comment]
I'm running it also after having the previous version installed.  The Jan 9 version attempted to start an onion service, but this one stops after bootstrap 100%.  It also says there's no torrc file. That needs to be fixed. 
```
Jan 27 02:12:22.737 [notice] Configuration file "C:\Users\techn\AppData\Roaming\VERGE\tor\torrc" not present, using reasonable defaults.
Jan 27 02:12:22.740 [warn] Path for GeoIPFile (<default>) is relative and will resolve to C:\Windows\system32\<default>. Is this what you wanted?
Jan 27 02:12:22.740 [warn] Path for GeoIPv6File (<default>) is relative and will resolve to C:\Windows\system32\<default>. Is this what you wanted?
```

**Comment by Technophant** [comment]
Delete your database and txleveldb first.  I notice that the old ones were causing it essentially hang. Ran again:

```
Jan 27 23:17:12.000 [notice] Bootstrapped 100%: Done
Jan 27 23:18:00.000 [notice] Closing stream for '[scrubbed].onion': hidden service is unavailable (try again later).
Jan 27 23:20:29.000 [warn] accept() failed: Successful WSAStartup not yet performed [WSANOTINITIALISED ]. Closing listener.
Jan 27 23:20:29.000 [err] libevent call with win32 failed: Successful WSAStartup not yet performed [WSANOTINITIALISED ] [10093]
```
Next tried deleting the onion folder and it assigned a new onion address. Got:
```
Jan 27 23:21:23.000 [notice] Starting with guard context "default"
Jan 27 23:21:23.000 [notice] Bootstrapped 80%: Connecting to the Tor network
```
I'm going to next try creating and editing a torrc file to see if I can get things working.

**Comment by knight9175** [comment]
@ben:

*result of running verged.exe in a cmd window: *

*Jan 27 21:26:55.934 [notice] Tor 0.3.2.8-rc-dev running on Very recent
version of Windows [major=10,minor=0] with Libevent 2.0.22-stable, OpenSSL
1.0.2k, Zlib 1.2.11, Liblzma N/A, and Libzstd N/A.*
*Jan 27 21:26:55.934 [notice] Tor can't help you if you use it wrong! Learn
how to be safe at https://www.torproject.org/download/download#warning
<https://www.torproject.org/download/download#warning>*
*Jan 27 21:26:55.935 [notice] Configuration file
"C:\Users\Alireza\AppData\Roaming\VERGE\tor\torrc" not present, using
reasonable defaults.*
*Jan 27 21:26:55.937 [warn] Path for GeoIPFile (<default>) is relative and
will resolve to D:\crypto\vergeWindows\<default>. Is this what you wanted?*
*Jan 27 21:26:55.937 [warn] Path for GeoIPv6File (<default>) is relative
and will resolve to D:\crypto\vergeWindows\<default>. Is this what you
wanted?*
*Jan 27 21:26:55.946 [warn] Failed to parse/validate config: If you set
UseBridges, you must specify at least one bridge.*
*Jan 27 21:26:55.946 [err] Reading config failed--see warnings above.*
*VERGE: Timed out waiting for onion hostname.*

**Comment by knight9175** [comment]
@jeffs

thanks that did create a tor directory, a database directory. and a few
other log files. but did not create an onion directory and resulted in the
same Error.

**Comment by Technophant** [comment]
Ok, have it working.  I did rename the VERGE folder, but what worked was going to security and added Everyone and made sure child folders inherited the permission.  The bootstrapping completed and I have connections.

**Comment by Technophant** [comment]
I'm trying to reproduce my results.  Created new Verge folder.  First run, timed out and close. No onion folder made. 2nd run, onion folder made, also timed out waiting for onion. tor.log written to. boostrap 100%.  3d run, tor connection made and connections are slowly added.

**Comment by jrsikors** [comment]
I noticed in previous logs that it only tries 8 times to 'find onion hostname'.  Might be a dumb suggestion, but would it help to increase this ?  Or delay the 'retry' ?  I noticed on mine (has been working fine) , I made the connection on the 6th try.  That means that I was pretty close to receiving this error as well?

_No onion hostname yet, will retry in 2 seconds... (1/8)
No onion hostname yet, will retry in 2 seconds... (2/8)
No onion hostname yet, will retry in 2 seconds... (3/8)
No onion hostname yet, will retry in 2 seconds... (4/8)
Onion thread exited.No onion hostname yet, will retry in 2 seconds... (5/8)
No onion hostname yet, will retry in 2 seconds... (6/8)
No onion hostname yet, will retry in 2 seconds... (7/8)
No onion hostname yet, will retry in 2 seconds... (8/8)_

**Comment by jrsikors** [comment]
so @knight9175  , if you want to try what @Technophant did ..
run verge-qt with a blank datadir...
so do a:
_verge-qt.exe -datadir=c:\vergetest\data**2**_

Win or lose -- it should happen quickly.  If you try it with a full blockchain, it's going to take ~10 minutes each time to open verge fully.  If you do it with no blockchain, it will open quickly.  This makes troubleshooting a lot quicker.

If the error pops up -- close out as soon as it opens .. and then re-open ... 
Try this about ~5 times to see if the onion error ever goes away.

If the error finally goes away -- you can do a couple things:
  A)  Always use verge-qt.exe -datadir=c:\vergetest\data2
  B)  Copy everything in c:\vergetest\data2 back to the <appdata>\roaming\verge default datadir ... then go download and/or unzip the blockchain to the datadir

**Comment by Technophant** [comment]
Try downloading the nightly blockchain and just using the peers.dat.  Clean install only has 7 and they are overloaded but it has a list of over 50 more. The file can't be read with a hex editor.  what format is it?  is it encrypted?

**Comment by jrsikors** [comment]
Gotcha!  The original topic here was about the "Onion Hostname not found" for @knight9175  .  His topic the other day got hijacked by a bunch of Mac people ... don't want this one to get overlooked too :smile: 

maybe make a new topic and keep the issues separate if you think it's important for the Verge devs to investigate... otherwise this one will get closed as soon as @knight9175  confirms he's good to go.

**Comment by knight9175** [comment]
I opened a command promt as Admin, did this 5 times with no luck.

firewall is down, Virus Protection is off, and I feel naked.

this is again the output of the verged.exe

Jan 28 10:24:34.406 [notice] Tor 0.3.2.8-rc-dev running on Very recent
version of Windows [major=10,minor=0] with Libevent 2.0.22-stable, OpenSSL
1.0.2k, Zlib 1.2.11, Liblzma N/A, and Libzstd N/A.
Jan 28 10:24:34.406 [notice] Tor can't help you if you use it wrong! Learn
how to be safe at https://www.torproject.org/download/download#warning
Jan 28 10:24:34.407 [notice] Configuration file
"C:\Users\Alireza\AppData\Roaming\VERGE\tor\torrc" not present, using
reasonable defaults.
Jan 28 10:24:34.409 [warn] Path for GeoIPFile (<default>) is relative and
will resolve to D:\crypto\vergeWindows\<default>. Is this what you wanted?
Jan 28 10:24:34.409 [warn] Path for GeoIPv6File (<default>) is relative and
will resolve to D:\crypto\vergeWindows\<default>. Is this what you wanted?
Jan 28 10:24:34.418 [warn] Failed to parse/validate config: If you set
UseBridges, you must specify at least one bridge.Jan 28 10:24:34.418 [err]
Reading config failed--see warnings above.
VERGE: Timed out waiting for onion hostname.

and @Jeffs thanks for getting this thread back on track.

**Comment by jrsikors** [comment]
Just to clarify -- your datadir is now D:\crypto\vergeWindows\

Is there a tor folder D:\crypto\vergeWindows\tor?  Is there a tor.log in that folder?  plz post the tor.log
Is there an onion folder in D:\crypto\vergeWindows\tor\onion ?  Are there 2 files generated in there?

Would it help solve @knight9175 if he just manually created the onion folder, and someone gave up their hostname and privatekey files?  I'm not savvy enough to know if the information in my 2 files are acceptable to give out.

**Comment by jrsikors** [comment]
clarifying my clarify post:
~~your datadir is now D:\crypto\vergeWindows\
~~you are running this: verge-qt.exe -datadir=D:\crypto\vergeWindows\

when you do that -- you are still getting a "Timed out waiting for Hostname" , and your application just straight quits.

Is there a tor folder D:\crypto\vergeWindows\tor? Is there a tor.log in that folder? plz post the tor.log

Is there an onion folder in D:\crypto\vergeWindows\tor\onion ? Are there 2 files generated in there?

**Comment by knight9175** [comment]
Yes. There is a verge Tor directory.
there is no tor.log file in it.
[image: Inline image 1]

no there is no \onion\, or files in the tor directory. it never gets
created.

**Comment by jrsikors** [comment]
is D: a local drive? or a jump/external drive?  shouldn't matter, but just curious

**Comment by Technophant** [comment]
@justinvforvendetta we need to get on this, proto.  electrum wallet too. hard fork has happened fork height is here already. 

More tor problems:
```
Jan 28 11:33:39.000 [notice] Closing stream for '[scrubbed].onion': hidden service is unavailable (try again later).
Jan 28 11:56:08.000 [notice] Closing stream for '[scrubbed].onion': hidden service is unavailable (try again later).
Jan 28 12:22:40.000 [notice] Your system clock just jumped 491 seconds forward; assuming established circuits no longer work.
Jan 28 12:22:43.000 [notice] Tor has successfully opened a circuit. Looks like client functionality is working.
Jan 28 12:51:02.000 [notice] Closing stream for '[scrubbed].onion': hidden service is unavailable (try again later).
Jan 28 12:55:54.000 [warn] Fetching v2 rendezvous descriptor failed. Retrying at another directory.
Jan 28 12:55:55.000 [warn] Fetching v2 rendezvous descriptor failed. Retrying at another directory.
Jan 28 12:55:57.000 [notice] Closing stream for '[scrubbed].onion': hidden service is unavailable (try again later).
```
My clock is fine, just checked it.  Did not start this time.  If the core wallet and daemon aren't working and it's a required update, where are we now?  I haven't tried compiling the latest source code.  Using the latest windows release.

**Comment by jrsikors** [comment]
@Technophant ,, pleasae go open a new issue .. you will probably get more attention :smile:

We need this to try to help @knight9175 with his problem.  

I think your findings are great and worthy of further discussion, but out of respect for @knight9175 .. let's try to keep this focused on his problems.

**Comment by Technophant** [comment]
call it what?  multiple issues?  where to start...

**Comment by Max-Meldrum** [comment]
@knight9175 Did you try what @Technophant  wrote?
" but what worked was going to security and added Everyone in advanced and made sure child folders inherited the permission."

If I understand correctly. The tor directory gets created (by the application, not you), but nothing else is created under the tor directory?

**Comment by Technophant** [comment]
Yes, but I'm not sure that's the issue.  peer.dat seems to be the problem file. If you try this, go to advanced security, add, add principal, type everyone, choose full control, check replace all child permiss. and apply

**Comment by knight9175** [comment]
you are asking me to give everyone full control over my computer and you
don't see a problem with this ?

**Comment by knight9175** [comment]
@Max, yes the data directory is empty when I start. By running what @Jeffs
said, I get the tor directory a few files and a database directory created
for me by the application.

**Comment by knight9175** [comment]
I did it, I don't like it.. and it still doesn't work :

This is the contents of the debug.log file.

VERGE version v4.0.1.0-794660a-funky-beta (2018-01-19 03:31:22 -0500)
Using OpenSSL version OpenSSL 1.0.2k  26 Jan 2017
Startup time: 01/28/18 19:32:00
Default data directory C:\Users\Alireza\AppData\Roaming\VERGE
Default mining hash algorithm: scrypt
Used data directory D:\crypto\VergeTest\data
scrypt: using scrypt-sse2 as built.
dbenv.open LogDir=D:\crypto\VergeTest\data\database
ErrorFile=D:\crypto\VergeTest\data\db.log
Bound to 127.0.0.1:21102
TOR thread started.
Using OBFS4.
No onion hostname yet, will retry in 2 seconds... (1/8)
No onion hostname yet, will retry in 2 seconds... (2/8)
No onion hostname yet, will retry in 2 seconds... (3/8)
No onion hostname yet, will retry in 2 seconds... (4/8)
No onion hostname yet, will retry in 2 seconds... (5/8)
No onion hostname yet, will retry in 2 seconds... (6/8)
Onion thread exited.No onion hostname yet, will retry in 2 seconds... (7/8)
No onion hostname yet, will retry in 2 seconds... (8/8)

at this point I think you can close this issue, I've spent way too much
time on this and will wait for a few versions of the wallet to advance
before adopting the wallet. I also noticed that the android wallet requires
multiple other applications like tor to be downloaded.

I'll lookforward to the day all this is bundled up into a single app.



for now I've used the 3.0 wallet to move all my stuff over to binance where
I have 2FA.

Thanks again for your help guys.
Ali

**Comment by jrsikors** [comment]
@knight9175   the permission thing is just for the folder structure.

However -- the fact that your verge-qt wallet is creating SOME directories leads me to believe that this wouldn't fix your problem.

It wouldn't make sense that Verge-qt has permission to create <datadir>\tor folder, but doesn't have permissions to create <datadir>\tor\onion

The only reason I can think of as to why \onion\ isn't getting created is because you aren't making that onion hostname connection.  If that connection happened, \onion\ folder would get created no problem, i'm 99.99% sure.

So -- yeah man I don't know what to tell you.  Since this isn't a widespread problem, it's definitely isolated to your computer.  

Last thoughts are:
~Your ISP is blocking it
~Some program on your computer is blocking it

Although I would like to see your problem to completion... I'm about out of ideas too , sorry man :(

**Comment by knight9175** [comment]
No problem. Valiant effort. Just know that you guys have upgraded me out of
a secure wallet.

So I hope there is a solution soon.

**Comment by Technophant** [comment]
Is that your problem that the onion dir is still not being created?  What version are running?  latest windows release?

**Comment by knight9175** [comment]
Yes 4.0.1

**Comment by jrsikors** [comment]
Check out his original post -- has the details ...

He is on VERGE version v4.0.1.0-794660a-funky-beta .. which is the latest...

This was his original problem.  I still think the onion folder/files aren't getting created because his machine can't find an onion hostname.

But i will let you take a stab it @Technophant  , i'll step aside :)

VERGE version v4.0.1.0-794660a-funky-beta (2018-01-19 03:31:22 -0500)
Using OpenSSL version OpenSSL 1.0.2k 26 Jan 2017
Startup time: 01/26/18 06:34:11
Default data directory C:\Users\Alireza\AppData\Roaming\VERGE
Default mining hash algorithm: scrypt
Used data directory C:\Users\Alireza\AppData\Roaming\VERGE
scrypt: using scrypt-sse2 as built.
dbenv.open LogDir=C:\Users\Alireza\AppData\Roaming\VERGE\database ErrorFile=C:\Users\Alireza\AppData\Roaming\VERGE\db.log
Bound to 127.0.0.1:21102
TOR thread started.
Using OBFS4.
No onion hostname yet, will retry in 2 seconds... (1/8)
No onion hostname yet, will retry in 2 seconds... (2/8)
No onion hostname yet, will retry in 2 seconds... (3/8)
No onion hostname yet, will retry in 2 seconds... (4/8)
Onion thread exited.No onion hostname yet, will retry in 2 seconds... (5/8)
No onion hostname yet, will retry in 2 seconds... (6/8)
No onion hostname yet, will retry in 2 seconds... (7/8)
No onion hostname yet, will retry in 2 seconds... (8/8)

**Comment by Technophant** [comment]
@knight9175, please share your tor.log and try fixing the permissions. Try also renaming the verge folder and start fresh and give it 2 hours before you close it again IF the tor log says 100% bootstrap.

**Comment by knight9175** [comment]
Please read my posts above

**Comment by Technophant** [comment]
I've read it, but without the tor.log it's not possible to know what's going on. Please go through the troubleshooting steps above or wait for the non-tor version to be release to be compiled and released.

**Comment by jrsikors** [comment]
His tor.log doesn't even get generated.  The app quits out as soon as onion hostname cannot be found... so peers.dat don't even come into play.  I don't think peers.dat is related to being able to connect to tor / find onion hostname.   The app literally shuts down on him and he cannot open it at all.  

The <datadir>\tor\ folder gets created .. but no tor.log gets generated (probably because the app quits on him and never fully loads)

Basically -- the problem is -- as soon as it gets to 8/8 fails of "no onion hostname found" ... the app just completely shuts down / exits / quits. 

So -- the only solution is to figure out how to make his COMPUTER able to find an onion hostname.  Whether it's ISP, Anti Virus, another wallet, or another app that's stopping his computer from finding an onion hostname is really the mystery here :smile:

**Comment by Technophant** [comment]
Could use a firewall to block some ports and test this out. Not at my computer ATM. 
There's been 63 replies to OP but no answer yet. I would hate to leave this unresolved.

**Comment by knight9175** [comment]
Sorry for the late replies it's just at this point it's not worth our
collective effort. Specially since i have a working wallet in binance. I
will wait for the pc wallet to mature a bit and play with the suggestions I
have read here on my time to see if i can resolve the issue. If this is not
a widespread problem then it's on my PC as Jeffs has said.


I would rather your time be spent on further development. Verge has a
future. Let's get there.


Thank you, but please close this ticket.

**Comment by Technophant** [comment]
I'm guessing somewhere deep into your network config settings (and there's many) there's an issue.  It was working for me on vpn because the tor service is your router and it doesn't filter ports usually. Perhaps a setting would not allow the tor protocol or the proxy server.  A lot of networking stuff happens inside windows, not the app.

**Comment by knight9175** [comment]
I am running ubiquiti equipment. With a usg router at the head of it. Basic
install and not a lot of custom stuff on it other than a vlan for my WiFi
network.

**Comment by deliriumv** [comment]
Hi @knight9175, I understand you have already moved on from this for a while but just wanted to leave this here as something to try to discount any network/machine issues.

One of the things I always do when having any kind of issue with a tor enabled application is install the tor browser and verify through it that i can connect and use tor. (This will discount or confirm any issue with tor connectivity from your machine/network).

I couldn't see this reading back up through this thread so not sure if you have already tried this yet.
https://www.torproject.org/projects/torbrowser.html.en

**Comment by Technophant** [comment]
I'm going to guess that knight is a power user and has installed other networking in the past and it's altered his winsock (windows socket protocol) settings.  Please reset them by command line and check back with us.  I don't like having things unresolved: https://iihelp.iinet.net.au/Resetting_Network_Adapter_%28Winsock_Reset%29

**Comment by Technophant** [comment]
Just for fun you can type "netsh winsock show helper" to see a list of you networking software libraries.  You can also try "sfc --scannow" to verify you system files.

**Comment by jrsikors** [comment]
You have to make a bat file or shortcut to run verge-qt.exe with the additional info...
example:
verge-qt.exe -datadir=c:\vergetest\data

Would put your chain on C:\vergetest\data ... so change c:\vergetest\data to whatever you want.

**Comment by justinvforvendetta** [comment]
@jhmblvd what was the exact issue with windows10

**Comment by jrsikors** [comment]
@justinvforvendetta he was getting a "timed out waiting for onion hostname" error .. and the app wouldn't start

**Comment by jrsikors** [comment]
@knight9175  is also having the same problem... and we have just about given up.  There's no tor logs generated, there's no nothing of value generated.

The 'problem' is that the Verge-QT app literally exits after it 'times out waiting for onion hostname' ... so no tor logs get generated, no additional debug information available.

Verge-qt tries to find onion hostname 8 times... and if it can't .. it just quits out.

**Comment by jrsikors** [comment]
I do think the best suggestion so far that hasn't been tried was from @deliriumv a few posts ago.

for @jhmblvd  and @knight9175 

Trying to get Tor installed /connected by itself completely separate from the Verge-QT might provide better information/insight into what the real problem is.  Perhaps you will get more detailed information as to why you can't connect.  Get Tor Browser working... and in theory, then Verge-QT should work fine.

> One of the things I always do when having any kind of issue with a tor enabled application is install the tor browser and verify through it that i can connect and use tor. (This will discount or confirm any issue with tor connectivity from your machine/network).
> 
> https://www.torproject.org/projects/torbrowser.html.en

**Comment by Technophant** [comment]
Another troubleshooting step I've tried is use firefox and set the proxy to 127.0.0.1 port 9089 and then you can test actual tor connection to see if it work.  It would be nice to use the same setup screen that tor browser uses to configure it and also include don't use it (clearnet) or connect to an already running tor browser.

**Comment by justinvforvendetta** [comment]
tor is working perfectly in the latest wallets.
